phac-nml/irida-next
# Fix search and sort to work simultaneously

## What does this PR do and why?
This PR fixes the use of the filter and sort functionality on the projects dashboard and project samples pages, where users can now use both the filter and sort functions together. Previously, issues arose where sorting would remove the filter, or filtering would remove the selected sort (already fixed in PR [306](https://github.com/phac-nml/irida-next/pull/306)), which was unintuitive from a user perspective.

This PR is built on top of the changes from PR [306](https://github.com/phac-nml/irida-next/pull/306) where filtering was fixed and now submits without losing focus on the filter input.

## Screenshots or screen recordings
The issue:
[Screencast from 2023-11-28 12:51:07 PM.webm](https://github.com/phac-nml/irida-next/assets/82407232/25033f24-4c41-480a-85f8-f216ce971c60)

Solution:
[Screencast from 2023-11-28 12:53:57 PM.webm](https://github.com/phac-nml/irida-next/assets/82407232/c07da39b-68cf-4029-8c20-9da7e3b5594e)


## How to set up and validate locally
Navigate to the projects dashboard, project samples page, and groups dashboard, and ensure the filtering and sorting works as expected. Groups dashboard currently only has sorting

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
